src/core/border_router/routing_manager.cpp

```diff
@@ -3767,6 +3814,30 @@ Error RoutingManager::PdPrefixManager::Process(const RouterAdvert::RxMessage &aM
     return error;
 }
 
+bool RoutingManager::PdPrefixManager::ProcessPrefixEntry(DiscoveredPrefixTable::Entry &aEntry)
```

Comment:
How about passing `aFavoredEntry` to this method? I think we can then do more checks here and make the `PdPrefixManager::Process()` simpler?